src/components/flat-table/flat-table.test.tsx

```diff
@@ -0,0 +1,1480 @@
+    </FlatTable>
+  );
+
+  expect(screen.getByTestId("ft-wrapper")).toHaveStyle("overflow-x: hidden");
+  expect(screen.getByTestId("flat-table-container")).toHaveStyle(
+    "overflow-x: auto"
+  );
+});
+
+describe("rounded corners are enabled", () => {
+  it("should have the execped border radius styling when no footer is rendered", () => {
+    render(
+      <FlatTable data-role="ft-wrapper">
+        <FlatTableHead>
+          <FlatTableRow>
+            <FlatTableHeader>heading one</FlatTableHeader>
+          </FlatTableRow>
+        </FlatTableHead>
+        <FlatTableBody>
+          <FlatTableRow>
+            <FlatTableCell>child one</FlatTableCell>
+          </FlatTableRow>
+        </FlatTableBody>
+      </FlatTable>
+    );
+    const wrapper = screen.getByTestId("ft-wrapper");
+
+    expect(wrapper).toHaveStyleRule(
+      "border-top-left-radius",
+      "var(--borderRadius100)"
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-top-right-radius",
+      "var(--borderRadius100)"
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-bottom-left-radius",
+      "var(--borderRadius100)"
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-bottom-right-radius",
+      "var(--borderRadius100)"
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-top-left-radius",
+      "var(--borderRadius100)",
+      {
+        modifier: `thead ${StyledFlatTableRow}:first-of-type th:first-of-type`,
+      }
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-top-right-radius",
+      "var(--borderRadius100)",
+      {
+        modifier: `thead ${StyledFlatTableRow}:first-of-type th:last-of-type`,
+      }
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-bottom-left-radius",
+      "var(--borderRadius100)",
+      {
+        modifier: `tbody ${StyledFlatTableRow}:last-of-type td:first-child`,
+      }
+    );
+    expect(wrapper).toHaveStyleRule(
+      "border-bottom-right-radius",
+      "var(--borderRadius100)",
+      {
+        modifier: `tbody ${StyledFlatTableRow}:last-of-type td:last-child`,
+      }
+    );
+  });
+
+  it("should override Pager's top border styling so it connects to the table when passed in `footer`,", () => {
+    render(
+      <FlatTable footer={<Pager data-role="pager" onPagination={() => {}} />}>
+        <FlatTableHead>
+          <FlatTableRow>
+            <td>heading one</td>
+          </FlatTableRow>
+        </FlatTableHead>
+        <FlatTableBody>
+          <FlatTableRow>
+            <FlatTableCell>item one</FlatTableCell>
+          </FlatTableRow>
+        </FlatTableBody>
+      </FlatTable>
+    );
+    const pager = screen.getByTestId("pager");
+
+    expect(pager).toHaveStyle({
+      borderTop: "none",
+      borderTopLeftRadius: 0,
+      borderTopRightRadius: 0,
+    });
+  });
+
+  it("should not apply any border-radius on the table wrapper an set to 0 on Pager when passed as `footer` and `hasStickyFooter` set,", () => {
```

Comment:
**nitpick:** I also _think_ this is the same as the test on Line 68.

Reply:
good spot